### ISSUE TITLE
bug: defcustom's :key-value should be :value-type

### ISSUE THREAD

**Issue by shirakami-chigusa** [meow-edit/meow#614]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

When I try to change custom values ​​using `setopt` in init.el, I get a warning.

>setopt also checks whether the value is valid for the user option. For instance, using setopt to set a user option defined with a number type to a string will signal an error. 

[GNU Emacs Lisp Reference Manual - setopt](https://www.gnu.org/software/emacs/manual/html_mono/elisp.html#index-setopt)

```emacs-lisp
  (setopt
   meow-selection-command-fallback
   '((meow-change . meow-change-char)
     (meow-kill . meow-kill-whole-line)
     (meow-cancel-selection . keyboard-quit)
     (meow-pop-selection . meow-pop-grab)
     (meow-beacon-change . meow-beacon-change-char)))
```

>⛔ Warning (emacs): Value '(((meow-change . meow-change-char) (meow-kill . meow-kill-whole-line) (meow-cancel-selection . keyboard-quit) (meow-pop-selection . meow-pop-grab) (meow-beacon-change . meow-beacon-change-char)))' for 'meow-selection-command-fallback' does not match type (alist :key-type (function :tag Command) :key-value (function :tag Fallback))

https://github.com/meow-edit/meow/blob/bc05b048d3582425717d114b837b8fe78d47a753/meow-var.el#L49-L58

I think `:key-value` is wrong and `:value-type` is correct.

[Composite Types (GNU Emacs Lisp Reference Manual)](https://www.gnu.org/software/emacs/manual/html_node/elisp/Composite-Types.html)

>(alist :key-type key-type :value-type value-type)

All `defcustom` defined in `meow-var.el` have the same problem.

**Comment by eshrh** [comment]
I can't actually reproduce this exact error... i think it's because the value-type defaults to "sexp" if not specified. What version of emacs are you using?

Anyway, good catch, thanks!

**Comment by shirakami-chigusa** [comment]
I'm using Emacs 29.3 (`setopt` is a feature implemented in emacs 29.1).
`defcustom` values from all other packages can be used without warning.

I think There is no type called `:key-value`.`value-type` is correct type. Is there the type `:key-value`?

[Composite Types (GNU Emacs Lisp Reference Manual)](https://www.gnu.org/software/emacs/manual/html_node/elisp/Composite-Types.html)

- [Code search results](https://github.com/search?q=%3Avalue-type+emacs&type=code)
  - You can find a lot of Emacs code that uses `value-type`.
[Code search results](https://github.com/search?q=%3Akey-value+emacs&type=code)
  - All I could find was the code used by meow and helm.

**Comment by shirakami-chigusa** [comment]
> i think it's because the value-type defaults to "sexp" if not specified. 

Ah, I found out why the warning appeared.

The code that causes the warning is as follows:

```
 (setopt meow-selection-command-fallback
   '((meow-change . meow-change-char)
     (meow-kill . meow-kill-whole-line)
     (meow-cancel-selection . keyboard-quit)
     (meow-pop-selection . meow-pop-grab)
     (meow-beacon-change . meow-beacon-change-char)))
 (:require meow)
```

The correct code is below.

```
 (:require meow)
 (setopt meow-selection-command-fallback
   '((meow-change . meow-change-char)
     (meow-kill . meow-kill-whole-line)
     (meow-cancel-selection . keyboard-quit)
     (meow-pop-selection . meow-pop-grab)
     (meow-beacon-change . meow-beacon-change-char)))
```

Because I used `setopt` before `require`, the `defcustom` type definitions were not loaded yet.

Thanks!
